src/api/users.ts

```diff
@@ -0,0 +1,114 @@
+import { User } from '../models/user.model';
+import { Response, Router } from 'express';
+import { Request } from '../typings';
+import { ValidatedRequest } from 'express-joi-validation';
+import { asyncUserBodySchema, baseUserBodySchema, BaseUserSchema, validator } from '../validators';
+import { BaseUserAttributes, UserAttributes } from '../types/user';
+import { Op } from 'sequelize';
+
+export const router = Router();
+
+router.param('id', async (req: Request, _, next, id) => {
+	console.log('>>>>>> entered param');
+	req.user = (await User.findByPk(id)) ?? undefined;
+
+	next();
+});
+
+router
+	.route('/:id')
+	.get((req: Request<{ id: string }, UserAttributes, {}, {}, Record<string, BaseUserAttributes>>, res: Response) => {
+		if (req.user) {
+			res.json(req.user);
+		} else {
+			res.status(404).json({ message: `User with id ${req.params.id} is not found` });
+		}
+	})
+	.put(
+		validator.body(baseUserBodySchema),
+		async (
+			req: Request<{ id: string }, UserAttributes, BaseUserAttributes, {}, Record<string, BaseUserAttributes>>,
+			res
+		) => {
+			if (req.user) {
+				const user = req.user;
+				const { login, password, age } = req.body;
+
+				user.updateDetails(login, password, age);
+				await user.save();
+
+				res.status(200).json(user);
+			} else {
+				res.status(404).json({ message: `User with id ${req.params.id} is not found` });
+			}
+		}
+	)
+	.delete(async (req: Request<{ id: string }, {}, {}, {}, Record<string, BaseUserAttributes>>, res) => {
+		if (req.user) {
+			const user = req.user;
+
+			// TODO: DO WE NEED TO ONLY MARK USER FOR DELETION?
+			await user.destroy();
```

Comment:
Yes, you don't need to specify any field TTL or create a job for user deletions

Reply:
done